src/feat/signal.cc

```diff
@@ -0,0 +1,117 @@
+// feat/signal.cc
+
+// Copyright 2015  Tom Ko
+
+// See ../../COPYING for clarification regarding multiple authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//  http://www.apache.org/licenses/LICENSE-2.0
+//
+// THIS CODE IS PROVIDED *AS IS* BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, EITHER EXPRESS OR IMPLIED, INCLUDING WITHOUT LIMITATION ANY IMPLIED
+// WARRANTIES OR CONDITIONS OF TITLE, FITNESS FOR A PARTICULAR PURPOSE,
+// MERCHANTABLITY OR NON-INFRINGEMENT.
+// See the Apache 2 License for the specific language governing permissions and
+// limitations under the License.
+
+#include "base/kaldi-common.h"
+#include "util/common-utils.h"
+#include "feat/signal.h"
+
+namespace kaldi {
+
+void ElementwiseProductOfFft(const Vector<BaseFloat> &a, Vector<BaseFloat> *b) {
+  for (int i = 0; i < a.Dim() / 2; i++) {
```

Comment:
It's good practice to use a temporary
e.g.
int32 num_fft_bins = a.Dim() / 2;
and do i < num_fft_bins, to avoid the function call being repeated.
Also, inside the loop, please replace your code with:
    // do complex multiplication b *= a.
    ComplexMul(a(2*i), a(2*i + 1), &(b(2*i)), &(b(2*i + 1)));
which will be clearer.